uci_ssh.go

```diff
@@ -0,0 +1,121 @@
+package uci
+
+import (
+	"bytes"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+)
+
+type SshTree struct {
+	host   string
+	config ssh.ClientConfig
+	client *ssh.Client
+	tree
+	sync.Mutex
+}
+
+func NewSshTree(username string, password string, host string) (t *SshTree) {
+	config := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(password),
+		},
+		// Non-production only
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         5 * time.Second,
+	}
+
+	t = &SshTree{
+		config: *config,
+		host:   host,
+	}
+	var err error
+	t.client, err = ssh.Dial("tcp", t.host+":"+"22", &t.config)
+	if err != nil {
+		panic(err.Error())
+	}
+	return
+}
+
+func (t *SshTree) Disonnect() (err error) {
+
+	err = t.client.Close()
+	return err
+}
+
+// Overwrite the default loadconfig!
+func (t *SshTree) loadConfig(name string) (err error) {
+	session, err := t.client.NewSession()
+	defer session.Close()
+	session.Stderr = os.Stderr
+	var body bytes.Buffer
+	session.Stdout = &body
+	path := filepath.Join(DefaultTreePath, name)
+	err = session.Run("cat " + path)
```

Comment:
I'd assume that even the smallest OpenWrt systems provide support for SCP.

The implementation is relatively easy, and I found this resource helpful in the past: https://web.archive.org/web/20170215184048/https://blogs.oracle.com/janp/entry/how_the_scp_protocol_works

Reply:
True. I added this now. 
It was more me being too lazy. But indeed the suggestions make absolute sense for security.

Reply:
By the looks of it, the `github.com/tmc/scp` package does only file uploads. For downloads, the same security concerns apply here. Again, consider malicious config names, https://go.dev/play/p/JqyBPwls9o9.

Reply:
I added a method now. This is working.